oeps/oep-0026-arch-xapi-integration.rst

```diff
@@ -0,0 +1,421 @@
+   cross-maintained by Open edX xAPI consumers (**Validator**).
+3. Routes real-time xAPI events to all registered consumers (**Router**).
+4. Provides an administration interface to configure and manage registered
+   consumers (**Admin**).
+
+Translator
+~~~~~~~~~~
+
+This layer will plug into the `Open edX Event Tracking`_ subsystem to listen to
+events of interest and translate them to xAPI's JSON format.
+
+TBD - The development team will assess whether we should use (and start owning)
+the already implemented (but no longer maintained) `xAPI Python Open Source Library`_. 
+
+Validator
+~~~~~~~~~
+
+The intention of this validation layer is to ensure we continue to support the
+expected xAPI formats by all participating Open edX xAPI consumers. In many ways,
+this is similar to what `consumer-driven contract testing`_ would enable and uses
+similar design principles. Essentially, a common validation library can be
+collaboratively maintained by participating consumers, including consuming adaptive
+engines. This ensures an end-to-end integration that can be maintained going forward.
+
+**Note**: Although xAPI specifies a standardized format, it is a low-level transaction
+schema and relies on higher-level "profiles" applied on top of it. So the profiles
+for specific Activities, Verbs, Contexts, etc used by Open edX need to be contractually
+supported.
+ 
+Router
+~~~~~~
+
+The router layer forwards xAPI events to interested consumers.  The complexity of this
+layer may increase over time as we support higher degrees of customization by consumers,
+such as filtering for only certain types of events and varying degrees of permissions
+of access to activities.
+
+For the first iteration, we need the following permissions:
+
+* Course restriction - certain consumers can access events only in certain courses.
+
+In the future, we may need the following:
+
+* User restriction - certain consumers can access all events for certain users.
+* Site restriction - certain consumers are limited to accessing events of certain sites.
+* Activity type restriction - certain consumers can access only certain types of events.
+
+Admin
+~~~~~
+
+The administration layer is responsible for allowing the Open edX site administrator
+to configure the list and permissions of registered xAPI event consumers.
+
+Eventually, the registry of consumers may become a self-service portal where consumers
+may initiate their request for access. However, initially, this can be a simple interface,
+possibly implemented in Django Admin.
+
+.. _Open edX Event Tracking: https://github.com/edx/event-tracking
+.. _Django App Plugin: https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/plugins/README.rst
+.. _xAPI Python Open Source Library: https://xapi.com/python-library/
+.. _consumer-driven contract testing: https://www.thoughtworks.com/radar/techniques/consumer-driven-contract-testing
```

Comment:
If there is no routing layer, the administration layer should be implemented in the proxy (aka routing component).

Reply:
See my question above - would it be clearer if I use the term "component" instead of "layer"?  I didn't mean to imply a specific implementation when using the term "layer".